CRM/Member/Form/MembershipRenewal.php

```diff
@@ -166,51 +162,81 @@ public function preProcess() {
     }
 
     CRM_Utils_System::setTitle(ts('Renew Membership'));
+    // there is a bunch of processing on price set id.  Make sure to retrieve it from
+    // submitteValues first when doing "submit".
+    if (!empty($this->_submitValues['price_set_id'])) {
+      $this->_priceSetId = $this->_submitValues['price_set_id'];
+    }
+
+    if (empty($this->_priceSetId)) {
+      $pair = CRM_Price_BAO_PriceSet::getLastPriceSetUsed($this->_id);
+      if ($pair != NULL) {
+        $this->_priceSetId = $pair['price_set_id'];
+        $this->assign('show_price_set', $pair['price_set_is_through_contribution']);
+      }
+    }
+    if (!empty($this->_submitValues['price_set_id'])) {
+      $this->assign('show_price_set', TRUE);
+    }
 
     parent::preProcess();
   }
 
   /**
+   * Determine whether the price set includes memberships of types other than the current membership organisation.
+   *
+   * @return bool
+   */
+  private function getPriceSetHasNewOrgs() {
```

Comment:
I not sure what we mean by switch to renewal_date, or if that comment applies anymore.  Ok, looking more closely, removing the "confusing" description does make it uncalled from what I can tell.  Removed in next commit.